clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,759 @@
+func identRegex(name string) string {
+	return `((?P<` + name + `>\w+)|(\"(?P<` + name + `_quote>.+)\"))`
+}
+
+// Get the value of a named InfluxQL identifier from a regex match map.
+// Returns empty string if no match
+func getIdentFromMatches(matches *map[string]string, name string) string {
+	if val, ok := (*matches)[name]; ok && val != "" {
+		return val
+	} else if val, ok := (*matches)[name+"_quote"]; ok && val != "" {
+		return val
+	}
+	return ""
+}
+
+// Create a regex match map from a regexp with named groups.
+// Returns nil if no match.
+func reSubMatchMap(r *regexp.Regexp, str string) *map[string]string {
+	match := r.FindStringSubmatch(str)
+	if match == nil {
+		return nil
+	}
+	subMatchMap := make(map[string]string)
+	for i, name := range r.SubexpNames() {
+		if i != 0 {
+			subMatchMap[name] = match[i]
+		}
+	}
+	return &subMatchMap
+}
+
+// the (?i) clause makes the regex match case-insensitive
+var insertIntoRegex string = `^(?i)INSERT(\s+)INTO(\s+)` + identRegex("db") + `(\.` + identRegex("rp") + `)?(\s+)(?P<point>.+)$`
+var insertRegex string = `^(?i)INSERT(\s+)(?P<point>.+)$`
+
+func (c Client) insert(cmd string) {
+	var db string
+	var rp string
+	var point string
+	insertRgx := regexp.MustCompile(insertRegex)
+	insertIntoRgx := regexp.MustCompile(insertIntoRegex)
+	insertMatches := reSubMatchMap(insertRgx, cmd)
+	insertIntoMatches := reSubMatchMap(insertIntoRgx, cmd)
+	if insertIntoMatches != nil {
+		db = getIdentFromMatches(insertIntoMatches, "db")
+		rp = getIdentFromMatches(insertIntoMatches, "rp")
+		point = getIdentFromMatches(insertIntoMatches, "point")
+	} else if !insertIntoRgx.Match([]byte(cmd)) && insertMatches != nil {
+		db = c.Database
+		rp = c.RetentionPolicy
+		point = getIdentFromMatches(insertMatches, "point")
+		if db == "" {
+			color.Red("Please run \"use <database>\" to run \"INSERT <point\"")
+			return
+		}
+	} else {
+		color.Red("Expected \"INSERT INTO <database>.<retention_policy> <point>\" OR \"INSERT <point>\".")
+		return
+	}
+
+	buf := bytes.Buffer{}
+	gzw := gzip.NewWriter(&buf)
+
+	_, err := gzw.Write([]byte(point))
+	gzw.Close()
+	if err != nil {
+		color.Red("Failed to gzip points")
+		return
+	}
+	writeReq := c.PostLegacyWrite(context.Background()).
+		Db(db).
+		Rp(rp).
+		Precision(c.Precision).
+		ContentEncoding("gzip").
+		Body(buf.String())
+
+	if err := writeReq.Execute(); err != nil {
+		color.Red("ERR: %v", err)
+		if c.Database == "" {
+			fmt.Println("Note: error may be due to not setting a database or retention policy.")
+			fmt.Println(`Please set a database with the command "use <database>"`)
+			return
+		}
+	}
+}
+
+type FormatType string
+type FormatFunc func(api.InfluxqlJsonResponse)
+
+var (
+	CsvFormat    FormatType = "csv"
+	JsonFormat   FormatType = "json"
+	ColumnFormat FormatType = "column"
+)
+
+func (c *Client) runAndShowQuery(query string) {
+	// TODO: guide users trying to use deprecated InfluxQL queries
```

Comment:
Is there an issue for this? I like putting issue links right into the TODO comment so it is clear when reading the code whether the TODO is stale or still live.

Reply:
Made an issue https://github.com/influxdata/influx-cli/issues/397